python2/koans/about_lists.py

```diff
@@ -22,69 +22,69 @@ def test_list_literals(self):
+        self.assertEqual([1, 2], nums)
 
         nums.append(333)
-        self.assertEqual([1, 2, __], nums)
+        self.assertEqual([1, 2, 333], nums)
 
     def test_accessing_list_elements(self):
         noms = ['peanut', 'butter', 'and', 'jelly']
 
-        self.assertEqual(__, noms[0])
-        self.assertEqual(__, noms[3])
-        self.assertEqual(__, noms[-1])
-        self.assertEqual(__, noms[-3])
+        self.assertEqual('peanut', noms[0])
+        self.assertEqual('jelly', noms[3])
+        self.assertEqual('jelly', noms[-1])
+        self.assertEqual('butter', noms[-3])
 
     def test_slicing_lists(self):
         noms = ['peanut', 'butter', 'and', 'jelly']
 
-        self.assertEqual(__, noms[0:1])
-        self.assertEqual(__, noms[0:2])
-        self.assertEqual(__, noms[2:2])
-        self.assertEqual(__, noms[2:20])
-        self.assertEqual(__, noms[4:0])
-        self.assertEqual(__, noms[4:100])
-        self.assertEqual(__, noms[5:0])
+        self.assertEqual(['peanut'], noms[0:1])
+        self.assertEqual(['peanut', 'butter'], noms[0:2])
+        self.assertEqual([], noms[2:2])
+        self.assertEqual(['and','jelly'], noms[2:20])
+        self.assertEqual([], noms[4:0])
+        self.assertEqual([], noms[4:100])
+        self.assertEqual([], noms[5:0])
 
     def test_slicing_to_the_edge(self):
         noms = ['peanut', 'butter', 'and', 'jelly']
 
-        self.assertEqual(__, noms[2:])
-        self.assertEqual(__, noms[:2])
+        self.assertEqual(['and', 'jelly'], noms[2:])
+        self.assertEqual(['peanut', 'butter'], noms[:2])
 
     def test_lists_and_ranges(self):
         self.assertEqual(list, type(range(5)))
-        self.assertEqual(__, range(5))
-        self.assertEqual(__, range(5, 9))
+        self.assertEqual([0 , 1, 2, 3, 4], range(5))
+        self.assertEqual([5, 6, 7, 8], range(5, 9))
 
     def test_ranges_with_steps(self):
-        self.assertEqual(__, range(5, 3, -1))
-        self.assertEqual(__, range(0, 8, 2))
-        self.assertEqual(__, range(1, 8, 3))
-        self.assertEqual(__, range(5, -7, -4))
-        self.assertEqual(__, range(5, -8, -4))
+        self.assertEqual([5, 4], range(5, 3, -1))
+        self.assertEqual([0, 2, 4, 6], range(0, 8, 2))
+        self.assertEqual([1, 4, 7], range(1, 8, 3))
+        self.assertEqual([5, 1, -3], range(5, -7, -4))
+        self.assertEqual([5, 1, -3, -7], range(5, -8, -4))
 
     def test_insertions(self):
         knight = ['you', 'shall', 'pass']
         knight.insert(2, 'not')
-        self.assertEqual(__, knight)
+        self.assertEqual(['you', 'shall', 'not', 'pass'], knight)
 
         knight.insert(0, 'Arthur')
-        self.assertEqual(__, knight)
+        self.assertEqual(['Arthur', 'you', 'shall', 'not', 'pass'], knight)
 
     def test_popping_lists(self):
         stack = [10, 20, 30, 40]
         stack.append('last')
 
-        self.assertEqual(__, stack)
+        self.assertEqual([10, 20, 30, 40, 'last'], stack)
```

Comment:
Q: Compare this with C++. Can you create a vector such as this on C++? Yes? How. No? Why?

Reply:
You can't because when you declare a vector in C++ you have to choose the type of the members, so you can't mix `int` with `string` for example

Reply:
That's correct.

However, just so you can say you heard about it, there are some esoteric data types in C++1X that allow you to store data of different types in the same variable, but their use is not recommended and they are not as easy to use as python variables: https://en.cppreference.com/w/cpp/utility/any